File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans = new ArrayList<>();
        Map<String,List<String>> map = new HashMap<>();
        int n = strs.length;

        for(String s : strs){
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedString = new String(charArray);
            List<String> list = map.getOrDefault(sortedString, new ArrayList<>());
            list.add(s);
            map.put(sortedString, list);
        }

        for(var it: map.entrySet()){
            ans.add(it.getValue());
        }

        return ans;
        
    }
}

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;
        //idea is all anagram if i sort then they become requal 
        // so i can group all anagram by sorting them using sort value as key 

        for(auto it: strs){
            string sortedIt = it;
            sort(sortedIt.begin(), sortedIt.end());
            map[sortedIt].push_back(it);
        }
        vector<vector<string>> ans;
        for(auto [key, value]: map){
            ans.push_back(value);
        }
        return ans;
    }
};